Update view-secrets references

elsesiy opened the pull request:
Hi,
as announced to @ahmetb [here](https://github.com/ahmetb/kubectl-extras/pull/12), I re-wrote the `view-secrets` plugin.

This PR changes the existing references to the new repo.

elsesiy commented:
/check-cla

ahmetb commented on plugins.md:
```diff
@@ -62,7 +62,7 @@ Name | Description | Stars
 [support-bundle](https://github.com/replicatedhq/troubleshoot) | Creates support bundles for off-cluster analysis | ![GitHub stars](https://img.shields.io/github/stars/replicatedhq/troubleshoot.svg?label=stars&logo=github)
 [tail](https://github.com/boz/kail) | Stream logs from multiple pods and containers using simple, dynamic source selection. | ![GitHub stars](https://img.shields.io/github/stars/boz/kail.svg?label=stars&logo=github)
-[view-secret](https://github.com/ahmetb/kubectl-extras) | Decode secrets | ![GitHub stars](https://img.shields.io/github/stars/ahmetb/kubectl-extras.svg?label=stars&logo=github)
+[view-secret](https://github.com/elsesiy/kubectl-view-secret) | Decode secrets | ![GitHub stars](https://img.shields.io/github/stars/elsesiy/kubectl-view-secret.svg?label=stars&logo=github)
```
This file is auto generated. You don’t need to edit. Make sure your manifest has the homepage set correctly is all.

elsesiy replied:
Okay, I reverted this change

ahmetb commented on plugins/view-secret.yaml:
```diff
@@ -3,19 +3,43 @@ kind: Plugin
+  version: "v0.1.0"
+  homepage: https://github.com/elsesiy/kubectl-view-secret
+  shortDescription: |
+    Kubernetes CLI plugin to decode Kubernetes secrets
```
Everything is a plugin in this repo & on krew search output. :)

elsesiy replied:
Done!

ahmetb commented on plugins/view-secret.yaml:
```diff
@@ -3,19 +3,43 @@ kind: Plugin
+  shortDescription: |
+    Kubernetes CLI plugin to decode Kubernetes secrets
+  description: |+2
+    Let's you conveniently base64 decode all key/value pairs in a given secret.
```
You can delete up to “base64...“

elsesiy replied:
Done :)

ahmetb commented on plugins/view-secret.yaml:
```diff
@@ -3,19 +3,43 @@ kind: Plugin
+  description: |+2
+    Let's you conveniently base64 decode all key/value pairs in a given secret.
+
+    $ kubectl view-secret <secret>
```
Out of curiosity: do you support decoding certain data.* keys through a cli arg?

elsesiy replied:
Currently not, I thought of adding this and just today someone actually asked for it so I probably ship it in the next version 👍

ahmetb commented:
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

This pull request **doesn't** seem to be a straightforward version bump. I'll have a human review this.

_Why wasn't this detected as a plugin version bump:_

>file plugins/view-secret.yaml is not a straightforward version bump: diff line unrecognized for version bumps: [+  homepage: https://github.com/elsesiy/kubectl-view-secret]

ahmetb commented:
+1 for version/syntax parity. We don't know how many users exist out there, but if we can get the same syntax (or an error that shows the new syntax gracefully) we should wait on flipping the plugin manifest (this PR).

elsesiy commented:
@corneliusweig @ahmetb My bad, I didn't even notice this because I didn't spend too much time trying to use the current version. In general I'm happy to add this but this somewhat changes how I anticipated using this plugin. Even though there might be instances where specifying the key makes sense, I see it more as an optional and not a mandatory attribute as this changes the workflow again to:
1. `kubectl view-secret <sec>`
2. Copy key
3. `kubectl view-secret <key from 2.>`

I wanted to achieve simplicity by removing the friction of having to specifically providing either `namespace` or `secret:key`. Should I rename this plugin and submit it as a separate plugin?

ahmetb commented:
We should keep the name the same, since I won't maintain view-secret anymore.

Is it not at all possible to accommodate something like:
- `kubectl view-secret <name>`
- `kubectl view-secret <name.subkey>`

at the same time? -n/--namespace should still exist separately.

elsesiy commented:
I think that works. Another approach could be to use `kubectl view-secret <name> --all`
and then preserve the rest of the functionality as described above.
Which one do you prefer?

ahmetb commented:
I don't follow, what does --all do?

elsesiy commented:
`--all/-a` would output all `key:value` pairs from the provided secret whereas `kubectl view-secret <name>` would just print out the keys/follows the existing flow. But I do like your suggestion with `secret.key` a lot so I'll probably go ahead with this one

ahmetb commented:
Why do you need —all? Existing plugin already prints all keys in a secret without needing any argument. Please try the existing plugin first. 🙃

elsesiy commented:
Yes I've seen the current plugin, `--all` however would print `keys=values` and not just `keys`. Anyway as I said, I'll go with the `name.key` approach!

ahmetb commented:
Yeah in case that makes sense. —all can print all secrets in the ns in k=v format.

corneliusweig commented:
So just to clarify to avoid misunderstandings about the semantics of `--all`:
@elsesiy you are going to reproduce the behavior of `view-secrets` where it will print a list of keys in a given secret, when called as `kubectl view-secret my-secret`. When called as `kubectl view-secret my-secret my-key`, it will decode the content under `my-key` in `my-secret`. That's the behavior of the old implementation.
In addition, you'll support a flag `--all`, which is used as `kubectl view-secret my-secret --all`. This will print all keys and decoded values inside a given secret.

Is that correct?

elsesiy commented:
@corneliusweig yes that's exactly how I implemented it now to ensure backward-compatibility but also allow for easy decoding of all contents of a secret. I'll publish it as `v0.2.0` and update this PR soon!

corneliusweig commented:
@elsesiy If you are revisiting your plugin, also consider if you really need the "k8s.io/cli-runtime/pkg/genericclioptions" and "k8s.io/utils/pointer" dependencies. You are not really making use of those, because you are shelling out to kubectl and the flags don't get passed along. I think you can make your binaries smaller by removing those deps (8MB compressed is quite hefty for what you are doing).

elsesiy commented:
@corneliusweig Yeah I noticed that too and removed these dependencies too, thanks!!

ahmetb commented:
/hold
Please let us know when you want to move forward with the new plugin. I'm putting this on hold since it seems like you wanna make some changes.

ahmetb commented:
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

This pull request **doesn't** seem to be a straightforward version bump. I'll have a human review this.

_Why wasn't this detected as a plugin version bump:_

>file plugins/view-secret.yaml is not a straightforward version bump: diff line unrecognized for version bumps: [+  homepage: https://github.com/elsesiy/kubectl-view-secret]

elsesiy commented:
@corneliusweig @ahmetb I just pushed the changes as discussed, let me know if you have any remarks :)

corneliusweig commented on plugins/view-secret.yaml:
```diff
@@ -3,19 +3,45 @@ kind: Plugin
+  description: |+2
+    Base64 decode by key or all key/value pairs in a given secret.
+
+    $ kubectl view-secret <secret> # print secret keys
```
nit: can you align the comments? That would read so much nicer...

ahmetb replied:
> 2\. The output is still not quite the same.

I think this is ok.

elsesiy replied:
Will do but I just noticed I'm also missing [this](https://github.com/ahmetb/kubectl-extras/blob/master/view-secret/view-secret.sh#L38) case here. Do you want me to add it as well?

ahmetb replied:
I think it would be nice. Many Secrets will only have 1 key.

Do this:
1. print these communication messages to `stderr`, print parseable output (by scripts etc) to `stdout`
2. offer a `-q`/`--quiet` to turn off communication messages like ("I'm selecting the only key").

elsesiy replied:
Done!

ahmetb commented:
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

This pull request **doesn't** seem to be a straightforward version bump. I'll have a human review this.

_Why wasn't this detected as a plugin version bump:_

>file plugins/view-secret.yaml is not a straightforward version bump: diff line unrecognized for version bumps: [+  homepage: https://github.com/elsesiy/kubectl-view-secret]

elsesiy commented:
@corneliusweig @ahmetb Alright, I changed the format to the original output format, log info messages to `stderr` and added the `-q/--quiet` flag. Please let me know if there's anything missing at this point!

corneliusweig commented:
@elsesiy great, this looks pretty good now! Thanks for bearing with us through this lengthy review round and for taking over maintenance of this plugin!

/approve
/lgtm

corneliusweig commented:
/hold cancel

elsesiy commented:
@corneliusweig Thank you, it was a good experience and sorry for not paying too much attention to the backwards-compatibility from the beginning! Won't happen again :) I think the `plugins.md` needs to be refreshed again, right?

ahmetb commented:
I'll send a PR to re-generate that, you can do as well if you like.
